### PR TITLE
Bug fix: Synonym Sync: Entries without exact mappings

### LINKED ISSUE #687: Bug: Synonym Sync: `-confirmed` has non-exact entries
There are two Mondo terms where the Synonym Sync pipeline adds source evidence for a confirmed synonym where the Synonym Sync pipeline should not. These Mondo terms from the qc check are: 
```
http://purl.obolibrary.org/obo/MONDO_0012817,Ewing sarcoma,NCIT:C7542,PNET of Thoracopulmonary region,MONDO:relatedTo
http://purl.obolibrary.org/obo/MONDO_0002514,hepatobiliary neoplasm,DOID:3117,hepatobiliary tumors,MONDO:mondoIsBroaderThanSource
```

These should be sorted out with high priority to resolve this issue, but it is not blocking the existing work on the Synonym Sync pipeline from being merged and getting into `main`. So #93 and #671 should be merged before working on this at this time.

### PR DESCRIPTION
**Pull request by joeflack4**
resolves #687

> [!note]
> - Draft while working on: #696

## Overview
The synonym sync template TSVs were being built incorrectly, and were sometimes adding synchronization outputs for cases where the Mondo term and source term had no exact mapping in `mondo.sssom.tsv`. The filtering was not being applied correctly.

The cause of these appearing in the first place is that these pairs of Mondo+source terms have synonyms on them, but they do not have exact mappings.

### Documentation
Was the documentation added/updated under `docs/`?
- [ ] Yes
- [x] No, updates to the docs were not necessary after careful consideration

### QC
Was the full pipeline run before submitting this PR using `sh run.sh make build-mondo-ingest` on this branch (after 
`docker pull obolibrary/odkfull:dev`), and no errors occurred?
- [x] Yes
- [ ] No, there are no functional (code-related) changes to the pipeline in the PR, so no re-run is necessary

**Mini-build:**   
Not a full build yet, but there is a mini-build which compares the only synonym sync outputs before and after.
- #696

Build:
- #707

### New Packages
Were any new *Python packages* added?
- [ ] Yes, [requirements.txt.full](https://github.com/INCATools/ontology-development-kit/blob/master/requirements.txt.full) was updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

Were any other *non-Python packages* added?
- [ ] Yes, [Dockerfile](https://github.com/INCATools/ontology-development-kit/blob/master/Dockerfile) updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

### PR Review and Conversations Resolved
Has the PR been sufficiently reviewed by at least 1 team member of the Mondo Technical team and all threads resolved?
- [ ] Yes

### CODE REVIEW

File: src/scripts/sync_synonym.py
[the remark this answers is not in the document]
Comment (matentzn) [reply src/scripts/sync_synonym.py:1]: Check discussion and consider to merge into the new PR

[the remark this answers is not in the document]
Comment (joeflack4) [reply src/scripts/sync_synonym.py:1]: I do prefer #697. It seems we'll likely use that one instead, and if so, I'll update remake this PR using that as a base instead. Thanks!

**Review by matentzn — APPROVED** [review]
Assuming (which I cant see in the diff) that `mappings_df` is only exact matches, this LGTM!

**Comment by joeflack4** [comment]
> Assuming... that `mappings_df` is only exact matches

Yep! in `mondo.sssom.tsv` that does seem the case, but it also does a filter for exact just in case.

**Review by twhetzel — APPROVED** [review]
Mainly approving on Nico's approval here (and a previous discussion where these code changes look like what was needed) since I have not had enough time to look at this and the results in as much detail as I would prefer.

**Comment by twhetzel** [comment]
@joeflack4 I'll this for you to merge

**Comment by joeflack4** [comment]
@twhetzel OK, I'll merge it now. I just got done reviewing the build #707 and it is fine, consistent with the preliminary mini-build #696.

**Pull request merged**